config-gen/apis/v1alpha1/types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"io/ioutil"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+)
+
+// KubebuilderProject implements the API for generating configuration
+type KubebuilderProject struct {
```

Comment:
What about it be something more generic such as ConfigGenerateProject instead ?

Reply:
I don't have a strong opinion.  `KubebuilderConfigGen` ?

Reply:
+1 for ` KubebuilderConfigGen`

Reply:
I was looking for we remove Kubebuilder from the name since it can be used in any place. however, not strong opiniated on that as well. I am fine with.

Reply:
I think having kubebuilder in the name will make it more clear what this is vs other files that may be on disk.  Since the command is part of kubebuilder (as written), I am not sure if there is much to be gained by loosely coupling the `kind` to kubebuilder.